### PR TITLE
Add support for labels in config and `--label` flag on CLI

### PR DESCRIPTION
**Pull request by autarch**
@oschwald If you could take a quick look at this, I'd appreciate it. @mmcclimon, you might find this of interest as well.

### CODE REVIEW

File: Changes.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 <!-- next-header -->
 
+## 0.5.1
+
+- Added a new labels feature. This allows you to group commands in your config
+  file by assigning one or more `labels` in their config. Then when running
+  `precius`, you can run commands with a specific label: `precious lint
```
Comment (mmcclimon) [inline Changes.md:7]: typo: `precius` -> `precious`

File: README.md [hunk 2]
```diff
@@ -268,6 +269,27 @@ The name passed to `--command` must match the name of the command in your
 config file. So in the above example, this would look for a command defined as
 `[commands.some-command]` in your config.
 
+#### Selecting Commands With Labels
+
+Each command can be assigned one or more labels. This lets you create arbitrary groups of
+command. Then when you tidy or lint you can pick a label by passing a
```
Comment (mmcclimon) [inline README.md:275]: Probably you want "arbitrary groups of commands" here.

File: precious-core/src/config.rs [hunk 3]
```diff
@@ -391,6 +402,8 @@ impl Config {
     }
 }
 
+const DEFAULT_LABEL: &str = "default";
```
Comment (mmcclimon) [inline precious-core/src/config.rs:405]: (nit) I would stick this above the `impl Config` line; I know rustc doesn't care but reading into_commands, I found myself wondering where it came from.

File: precious-core/src/config.rs [hunk 4]
```diff
@@ -850,11 +870,54 @@ mod tests {
...
         let err = res.unwrap_err().downcast::<ConfigError>().unwrap();
         assert_eq!(err, expect_err);
 
         Ok(())
     }
+
+    #[test_case(vec![], "default", true)]
+    #[test_case(vec!["default".to_string()], "default", true)]
+    #[test_case(vec!["default".to_string(), "foo".to_string()], "default", true)]
+    #[test_case(vec!["default".to_string(), "foo".to_string()], "foo", true)]
```
Comment (mmcclimon) [inline precious-core/src/config.rs:885]: I might also add a test case for only-non-default label matching:

```
    #[test_case(vec!["foo".to_string()], "foo", true)]
```

**Review by mmcclimon — COMMENTED** [review]
Yeah, nice. I think this is better than the thing I was working on, and requires less mucking around in the guts to boot.

File: Changes.md (hunk 1 above)
Comment (autarch) [reply Changes.md]: Thanks! I fixed this and all the other things you noted.

**Comment by oschwald** [comment]
This looks good to me! I could see a flag to allow running everything being useful, but that could be added later.

**Pull request merged**